### PR TITLE
feat(TBD-5959): Support both Standard and Legacy SQL for BigQuery

### PR DESCRIPTION
**Pull request by kcoepeau**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by kcoepeau** [comment]
Official default is Standard SQL so why not use it as default and keep old job using Legacy as is with a migration task. And provide the option for everyone to choose to use Legacy.

File: main/plugins/org.talend.designer.components.localprovider/components/tBigQueryBulkExec/tBigQueryBulkExec_begin.javajet [hunk 1]
```diff
@@ -470,15 +471,15 @@
 						String columnName = column.getLabel();
 						String typeToGenerate = null;
 						if("id_String".equals(column.getTalendType())) {
-							typeToGenerate = "com.google.cloud.bigquery.LegacySQLTypeName.STRING";
+							typeToGenerate = useLegacySql ? "com.google.cloud.bigquery.LegacySQLTypeName.STRING" : "com.google.cloud.bigquery.StandardSQLTypeName.STRING";
```
Comment (yu-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tBigQueryBulkExec/tBigQueryBulkExec_begin.javajet]: The method calling this `com.google.cloud.bigquery.Field.of` is for `Legacy SQL`, and not applicable for `Standard SQL`, need to changed here.

**Comment by kcoepeau** [comment]
@yu-talend reverted type selection following this documentation that still uses LagacyTypes : https://cloud.google.com/bigquery/docs/tables

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/AddUseLegacySqlForBigQueryTask.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        private String field = "CHECK"; //$NON-NLS-1$
+
+        private String name = "USE_LEGACY_SQL"; //$NON-NLS-1$
+
+        public void transform(NodeType node) {
+
+            if (ComponentUtilities.getNodeProperty(node, name) == null) {
+                ComponentUtilities.addNodeProperty(node, name, field);
+                ComponentUtilities.setNodeValue(node, name, "false");
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/AddUseLegacySqlForBigQueryTask.java]: Shouldn't it be true ? (legacy SQL used for old jobs)

Comment (kcoepeau) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/AddUseLegacySqlForBigQueryTask.java]: For the types, the schema builder only use legacy types and there is no schema builder that use the standard type, also the official documentation uses legacy types in any case.

**Pull request merged**